src/core/thread/mle_router.cpp

```diff
@@ -374,6 +374,12 @@ void MleRouter::SetStateLeader(uint16_t aRloc16)
     otLogNoteMle("Leader partition id 0x%x", mLeaderData.GetPartitionId());
 }
 
+void MleRouter::AddLeaderAloc(void)
+{
+    IgnoreError(GetLeaderAloc(mLeaderAloc.GetAddress()));
+    IgnoreError(Get<ThreadNetif>().AddUnicastAddress(mLeaderAloc));
+}
```

Comment:
I verified that we don't need `VerifyOrExit(IsLeader())`
Noticed it is being used only from `SetStateLeader()`. Can we consider removing this method (inlining it)?